src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -211,16 +212,13 @@ public void getAlerts(@NonNull Date startTime, @NonNull Date finishTime, @NonNul
 
     public void acknowledgeAlert(@NonNull Alert alert,
                                  @NonNull Callback<List<String>> callback) {
-        URI uri = Uris.getUri(String.format(BackendPipes.Paths.ALERT_ACKNOWLEDGE, alert.getId()));
-
-        readPipe(BackendPipes.Names.ALERT_ACKNOWLEDGE, uri, callback);
+        alert.setId(alert.getId());
```

Comment:
That sounds absolutely wrong. The IDE should not force you to do a code that is not needed. If there's a code optimization somewhere that is removing this, there should be an option to disable this "code eating" stuff. In any case, the solution is *not* to pollute the code with unnecessary code :)